lib/fastlane/plugin/ddg_apple_automation/actions/asana_add_comment_action.rb

```diff
@@ -0,0 +1,130 @@
+          FastlaneCore::ConfigItem.new(key: :task_url,
+                                       description: "Asana task URL",
+                                       optional: true,
+                                       type: String),
+          FastlaneCore::ConfigItem.new(key: :comment,
+                                       description: "Comment to add to the Asana task",
+                                       optional: true,
+                                       type: String),
+          FastlaneCore::ConfigItem.new(key: :template_name,
+                                       description: "Name of a template file (without extension) for the comment. Templates can be found in assets/asana_add_comment/templates subdirectory.
+      The file is processed before being sent to Asana",
+                                       optional: true,
+                                       type: String),
+          FastlaneCore::ConfigItem.new(key: :workflow_url,
+                                       description: "Workflow URL to include in the comment",
+                                       optional: true,
+                                       type: String)
+        ]
+      end
+
+      def self.is_supported?(platform)
+        true
+      end
+
+      def self.validate_params(task_id, task_url, comment, template_name, workflow_url)
+        if task_id.nil? && task_url.nil?
+          raise ArgumentError, "Both task_id and task_url cannot be nil. At least one must be provided."
+        end
+
+        if comment.nil? && template_name.nil?
+          raise ArgumentError, "Both comment and template_name cannot be nil. At least one must be provided."
+        end
+
+        if comment && workflow_url.nil?
+          raise ArgumentError, "If comment is provided, workflow_url cannot be nil"
+        end
+      end
+
+      def self.load_template_file(template_name)
+        template_file = Helper::DdgAppleAutomationHelper.load_asset_file("asana_add_comment/templates/#{template_name}.yml")
+        File.read(template_file)
+      rescue StandardError
+        UI.user_error!("Error: The file '#{template_name}.yml' does not exist.")
+        nil
+      end
+
+      def self.create_story(asana_access_token, task_id, text: nil, html_text: nil)
+        client = Asana::Client.new do |c|
+          c.authentication(:access_token, asana_access_token)
+        end
+        begin
+          if text
+            client.stories.create_story_for_task(task_gid: task_id, text: text)
+          else
+            client.stories.create_story_for_task(task_gid: task_id, html_text: html_text)
+          end
+        rescue StandardError => e
+          UI.user_error!("Failed to post comment: #{e}")
+        end
+      end
+
+      def self.process_template_content(template_content)
+        processed_content = template_content.gsub(/\$\{(\w+)\}/) { ENV.fetch($1, '') }
+        processed_content.gsub(/\s*\n\s*/, ' ').strip
```

Comment:
I think the following would be equivalent:

```suggestion
        template_content.gsub(/\$\{(\w+)\}/) { ENV.fetch($1, '') }
                        .gsub(/\s*\n\s*/, ' ')
                        .strip
```